app/src/main/java/de/tum/in/tumcampusapp/api/tumonline/TUMOnlineClient.kt

```diff
@@ -0,0 +1,147 @@
+package de.tum.`in`.tumcampusapp.api.tumonline
+
+import android.content.Context
+import com.tickaroo.tikxml.TikXml
+import com.tickaroo.tikxml.retrofit.TikXmlConverterFactory
+import de.tum.`in`.tumcampusapp.api.app.Helper
+import de.tum.`in`.tumcampusapp.api.tumonline.interceptors.AddTokenInterceptor
+import de.tum.`in`.tumcampusapp.api.tumonline.interceptors.CacheResponseInterceptor
+import de.tum.`in`.tumcampusapp.api.tumonline.interceptors.CheckErrorInterceptor
+import de.tum.`in`.tumcampusapp.api.tumonline.interceptors.CheckTokenInterceptor
+import de.tum.`in`.tumcampusapp.api.tumonline.model.AccessToken
+import de.tum.`in`.tumcampusapp.api.tumonline.model.TokenConfirmation
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.CalendarItem
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.CreateEventResponse
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.DeleteEventResponse
+import de.tum.`in`.tumcampusapp.component.tumui.calendar.model.Events
+import de.tum.`in`.tumcampusapp.component.tumui.grades.model.ExamList
+import de.tum.`in`.tumcampusapp.component.tumui.lectures.model.LectureAppointmentsResponse
+import de.tum.`in`.tumcampusapp.component.tumui.lectures.model.LectureDetailsResponse
+import de.tum.`in`.tumcampusapp.component.tumui.lectures.model.LecturesResponse
+import de.tum.`in`.tumcampusapp.component.tumui.person.model.Employee
+import de.tum.`in`.tumcampusapp.component.tumui.person.model.IdentitySet
+import de.tum.`in`.tumcampusapp.component.tumui.person.model.PersonList
+import de.tum.`in`.tumcampusapp.component.tumui.tutionfees.model.TuitionList
+import de.tum.`in`.tumcampusapp.utils.CacheManager
+import de.tum.`in`.tumcampusapp.utils.Const
+import de.tum.`in`.tumcampusapp.utils.DateTimeUtils
+import retrofit2.Call
+import retrofit2.Retrofit
+
+class TUMOnlineClient(private val apiService: TUMOnlineAPIService) {
+
+    fun getCalendar(force: Boolean = false): Call<Events> {
```

Comment:
I really don't like boolean parameters.
From looking at a function call, one can't tell WTH this means.

```
getCalendar(true);
```
Without looking at the function definition, that parameter could be anything.
Please consider a proper Enum as parameter, which might even directly reference the underlying String.

```
getCalendar(CacheControl.doNotCache)
```
Is much nicer to read

Reply:
Android Studio displays the parameter name, but I agree with your feedback. Will adapt this accordingly.